File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/ColorPhotoDao.java
package ru.magflayer.spectrum.data.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import ru.magflayer.spectrum.data.entity.ColorPhoto;

@Dao
public interface ColorPhotoDao {

    @Query("SELECT * FROM ColorPhoto")
    List<ColorPhoto> loadPhotos();

    @Query("SELECT * FROM ColorPhoto WHERE file_path LIKE :path")
    ColorPhoto loadPhoto(String path);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long savePhoto(ColorPhoto colorPhoto);

    @Delete
    int deletePhoto(ColorPhoto colorPhoto);

}

File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/ColorPhotoDao.kt
package ru.magflayer.spectrum.data.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import ru.magflayer.spectrum.data.entity.ColorPhoto

@Dao
interface ColorPhotoDao {

    @Query("SELECT * FROM ColorPhoto")
    suspend fun loadPhotos(): List<ColorPhoto>

    @Query("SELECT * FROM ColorPhoto WHERE file_path LIKE :path")
    suspend fun loadPhoto(path: String): ColorPhoto

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun savePhoto(colorPhoto: ColorPhoto): Long

    @Delete
    suspend fun deletePhoto(colorPhoto: ColorPhoto): Int
}
